### ISSUE TITLE
[API] [Cart] Guest order error

### ISSUE THREAD

**Issue by oallain** [Sylius/Sylius#12023]
**Sylius version affected**: dev-master

**Description**  
The endpoint `GET ​/new-api​/shop​/orders​/{id}` throw an error when there is more than one guest cart in the database.

**Steps to reproduce**  
1. Go to https://master.demo.sylius.com/new-api
2. Create a cart with `POST ​/new-api​/shop​/orders`
3. Create a second cart
4. Try to get cart with `GET ​/new-api​/shop​/orders​/{id}` with the `tokenValue` of second cart
Error :
```
{
  "code": 500,
  "message": "More than one result was found for query although one row or none was expected."
}
```

**Possible Solution**  
The error is due to the request :
```
SELECT o, shippingAddress_a1, billingAddress_a2, items_a3 
FROM App\Entity\Order\Order o 
LEFT JOIN o.customer customer 
LEFT JOIN customer.user user 
LEFT JOIN o.shippingAddress shippingAddress_a1 
LEFT JOIN o.billingAddress billingAddress_a2 
LEFT JOIN o.items items_a3 
WHERE (o.tokenValue = :id_tokenValue AND user IS NULL) OR o.customer IS NULL
```
More exactly to the part `OR o.customer IS NULL` .
This is introduice by https://github.com/Sylius/Sylius/blob/79e2edfe745a16d73d928d3c7f8667c3dc504196/src/Sylius/Bundle/ApiBundle/Doctrine/QueryItemExtension/OrderGetMethodItemExtension.php#L71

I don't know the need for "orWhere" so I currently don't have a solution.

PS : Same is here for PATCH endpoints : https://github.com/Sylius/Sylius/blob/79e2edfe745a16d73d928d3c7f8667c3dc504196/src/Sylius/Bundle/ApiBundle/Doctrine/QueryItemExtension/OrderMethodsItemExtension.php#L88

**Comment by lchrusciel** [comment]
Hey Olivier, 

thanks for reporting. We will take care of it.

**Comment by Tomanhez** [comment]
I think that this is solution: https://github.com/Sylius/Sylius/pull/11895

**Comment by oallain** [comment]
Yes, i think too :+1:

**Comment by mespinosaz** [comment]
I can confirm, we had the same issue and I could fix it by making same change as #11895
